src/backend/services/docker_service.py

```
@@ -1,20 +1,82 @@
 import os
+import re
 import docker
 import logging
 import traceback
 import xml.etree.ElementTree as ET
 from typing import Generator, Dict, Any
```

Ruff (0.15.5)

[warning] Import from `collections.abc` instead: `Generator`

Import from `collections.abc`

(UP035)